recipes/lora_finetune_single_device.py

```diff
@@ -779,6 +788,38 @@ def train(self) -> None:
                     )
                 )
 
+                # VALIDATION LOOP
+                self._sampler_val.set_epoch(curr_epoch)
```

Comment:
At least as is it's not clear to me why we want to set the epoch on the val sampler. If we're running over the entire val dataloader once why does the order matter? (And if we're not this will make the val loss noisier, right? Is that what we want?)

Reply:
If running all over it, you are right, it doesn't make sense. When running only on a subset of the validation set, it might be interesting from a statistical perspective. In fact, testing with different random samples gives a more robust signal of proper generalization from the model. The argument against it is that it makes debugging difficult. For instance, you have some spike in the validation loss in your batch. Without random sampling, it’s easier to investigate it. Considering everything, I removed it.